### ISSUE TITLE
Add type annotation syntax for lambdas

### ISSUE THREAD

**Issue by j-hui** [ssm-lang/sslang#67]
Right now, if we want to type-annotate a lambda, e.g., `fun x { x }`, we need to write it postfix, i.e.,

```
(fun x
  x): a -> a
```

However, this can get kind of cumbersome. @sedwards-lab suggested that to keep the syntax uniform, we should also support the following:

```
fun (x: a) -> a
  x
fun x : a -> a
  x
```

This should of course generalize to lambdas with multiple arguments.

**Comment by sedwards-lab** [comment]
Yes. And it should be done using the same machinery somehow. `let f x y z = ...` and `let f = fun x y z -> ...` should be treated identically. Should we do the desugaring before understanding the type annotations?

**Comment by j-hui** [comment]
They are separate in the AST, but are desugared while lowering to IR.

What do you mean by "understanding" the type annotations?
